rust/otap-dataflow/crates/pdata/src/payload.rs

```diff
@@ -1,86 +1,6 @@
 // Copyright The OpenTelemetry Authors
 // SPDX-License-Identifier: Apache-2.0
 
-//! Implementation of the pipeline data that is passed between pipeline components.
-//!
-//! Internally, the data can be represented in the following formats:
-//! - OTLP Bytes - contain the OTLP service request messages serialized as protobuf
-//! - OTAP Arrow Bytes - the data is contained in `BatchArrowRecords` type which
-//!   contains the Arrow batches for each payload, serialized as Arrow IPC. This type is
-//!   what we'd receive from the OTAP GRPC service.
-//! - OTAP Arrow Records - the data is contained in Arrow `[RecordBatch]`s organized by
-//!   for efficient access by the arrow payload type.
-//!
-//! This module also contains conversions between the various types using the `From`
-//! and `TryFrom` traits. For example:
-//! ```
-//! # use std::sync::Arc;
-//! # use arrow::array::{RecordBatch, UInt16Array};
-//! # use arrow::datatypes::{DataType, Field, Schema};
-//! # use otap_df_pdata::otap::{OtapArrowRecords, Logs};
-//! # use otap_df_pdata::proto::opentelemetry::{
-//!     arrow::v1::ArrowPayloadType,
-//!     collector::logs::v1::ExportLogsServiceRequest,
-//!     common::v1::{AnyValue, InstrumentationScope, KeyValue},
-//!     logs::v1::{LogRecord, ResourceLogs, ScopeLogs, SeverityNumber},
-//!     resource::v1::Resource
-//! };
-//! # use otap_df_pdata::{OtapPayload, OtlpProtoBytes};
-//! # use prost::Message;
-//! let otlp_service_req = ExportLogsServiceRequest::new(vec![
-//!    ResourceLogs::new(
-//!        Resource::default(),
-//!        vec![
-//!            ScopeLogs::new(
-//!                InstrumentationScope::default(),
-//!                vec![
-//!                    LogRecord::build()
-//!                        .time_unix_nano(2u64)
-//!                        .severity_number(SeverityNumber::Info)
-//!                        .event_name("event")
-//!                        .attributes(vec![KeyValue::new("key", AnyValue::new_string("val"))])
-//!                        .finish(),
-//!                ],
-//!            ),
-//!        ],
-//!    ),
-//!  ]);
-//! let mut buf = Vec::new();
-//! otlp_service_req.encode(&mut buf).unwrap();
-//!
-//! // Create a new OtapPayload from OTLP bytes
-//! let payload: OtapPayload = OtlpProtoBytes::ExportLogsRequest(buf).into();
```

Comment:
This documentation is mainly targeting the `OtapPayload` enum which is defined in this file. Maybe we should remove the documentation from `pdata.rs` and not here?

Reply:
ah you're right. let me make that change -- good catch @utpilla

Reply:
@utpilla fixed!